### PR TITLE
feat: filter memberships for same member to render highest permission only

### LINKED ISSUE #729: Duplicate membership in child item
If you have a membership on the parent and the child, both will be display in the membership table. Probably we only want to display the closest one.

### PR DESCRIPTION
**Pull request by LinaYahya**
closes #729

### CODE REVIEW

File: src/components/item/sharing/ItemSharingTab.tsx [hunk 1]
```diff
@@ -33,6 +34,28 @@ type Props = {
...
+  for (const membership of memberships) {
+    const { member, permission } = membership;
+
+    if (
+      !permissionMap[member.id] ||
+      PermissionLevelCompare.gt(permission, permissionMap[member.id].permission)
+    ) {
+      permissionMap[member.id] = membership;
+    }
+  }
```
Comment (pyphilia) [inline src/components/item/sharing/ItemSharingTab.tsx:55]: You can optimize with a `.reduce` function. I guess the accumulator could look like `{...acc, [member.id]: membership }`

**Review by pyphilia — APPROVED** [review]

File: src/components/item/sharing/ItemSharingTab.tsx [hunk 2]
```diff
@@ -33,6 +34,27 @@ type Props = {
...
 };
 
+interface PermissionMap {
+  [key: string]: ItemMembership;
+}
+
+const reduceMembershipsForSameMember = (
+  memberships: ItemMembership[],
+): ItemMembership[] => {
+  const permissionMap = memberships.reduce((acc: PermissionMap, curr) => {
```
Comment (spaenleh) [inline src/components/item/sharing/ItemSharingTab.tsx:44]: FYI, this form also works, both are equivalent:
```suggestion
  const permissionMap = memberships.reduce<PermissionMap>((acc, curr) => {
```

File: src/components/item/sharing/ItemSharingTab.tsx [hunk 3]
```diff
@@ -33,6 +34,27 @@ type Props = {
   memberships?: ItemMembership[];
 };
 
+interface PermissionMap {
+  [key: string]: ItemMembership;
+}
+
+const reduceMembershipsForSameMember = (
```
Comment (spaenleh) [inline src/components/item/sharing/ItemSharingTab.tsx:41]: Nitpicking: I would change the name of this function to something like: `selectHighestMemberships`. The `ForSameMember` feels a bit weird. It should read naturally when you use it in a component prop.

**Review by spaenleh — APPROVED** [review]
Looks good. I just provide some comments.

**Review by spaenleh — APPROVED** [review]

**Review by pyphilia — APPROVED** [review]

**Pull request merged**